components/public-api-server/pkg/auth/middleware.go

```diff
@@ -11,25 +11,50 @@ import (
 	"github.com/bufbuild/connect-go"
 )
 
-// NewServerInterceptor creates a server-side interceptor which validates that an incoming request contains a Bearer Authorization header
-func NewServerInterceptor() connect.UnaryInterceptorFunc {
-	interceptor := func(next connect.UnaryFunc) connect.UnaryFunc {
-		return connect.UnaryFunc(func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+type AuthInterceptor struct {
+	accessToken string
+}
 
-			if req.Spec().IsClient {
-				return next(ctx, req)
-			}
+func (a *AuthInterceptor) WrapUnary(next connect.UnaryFunc) connect.UnaryFunc {
+	return connect.UnaryFunc(func(ctx context.Context, req connect.AnyRequest) (connect.AnyResponse, error) {
+		if req.Spec().IsClient {
+			ctx = TokenToContext(ctx, NewAccessToken(a.accessToken))
+
+			req.Header().Add(authorizationHeaderKey, bearerPrefix+a.accessToken)
+			return next(ctx, req)
+		}
 
-			token, err := tokenFromRequest(ctx, req)
-			if err != nil {
-				return nil, err
-			}
+		token, err := tokenFromRequest(ctx, req)
+		if err != nil {
+			return nil, err
+		}
 
-			return next(TokenToContext(ctx, token), req)
-		})
+		return next(TokenToContext(ctx, token), req)
+	})
+}
+
+func (a *AuthInterceptor) WrapStreamingClient(next connect.StreamingClientFunc) connect.StreamingClientFunc {
+	return func(ctx context.Context, s connect.Spec) connect.StreamingClientConn {
+		ctx = TokenToContext(ctx, NewAccessToken(a.accessToken))
+		conn := next(ctx, s)
+		conn.RequestHeader().Add(authorizationHeaderKey, bearerPrefix+a.accessToken)
+		return conn
 	}
+}
 
-	return connect.UnaryInterceptorFunc(interceptor)
+func (a *AuthInterceptor) WrapStreamingHandler(next connect.StreamingHandlerFunc) connect.StreamingHandlerFunc {
+	return func(ctx context.Context, conn connect.StreamingHandlerConn) error {
+		token, err := tokenFromConn(ctx, conn)
+		if err != nil {
+			return err
+		}
+		return next(TokenToContext(ctx, token), conn)
+	}
+}
```

Comment:
Please add unit tests for these

Reply:
@easyCZ  Do you have a pointer how to do it for the streaming interceptor? there's no handy function like `connect.NewRequest` used for the unary interceptor

Reply:
I think the easiest is to pick some proto which we already have, and test it on a real RPC from a server which we setup in the tests and a client which invokes that server.

Mostly what we're after is that the context makes it into the handler. Let me know if you'd like a hand with this.

Reply:
Added the test, let me know if it makes sense